lib/trie/hash.go

```diff
@@ -17,31 +17,64 @@
 package trie
 
 import (
+	"bytes"
+	"context"
 	"hash"
+	"sync"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/scale"
+	"golang.org/x/sync/errgroup"
 
 	"golang.org/x/crypto/blake2b"
 )
 
+type sliceBuffer []byte
+
+func (b *sliceBuffer) Write(data []byte) (n int, err error) {
+	*b = append(*b, data...)
+	return len(data), nil
+}
+
+func (b *sliceBuffer) Reset() {
+	*b = (*b)[:0]
+}
+
 // Hasher is a wrapper around a hash function
 type Hasher struct {
-	hash hash.Hash
+	hash     hash.Hash
+	tmp      sliceBuffer
+	parallel bool // Whether to use parallel threads when hashing
+}
+
+// HasherPool creates a pool of Hasher.
+var HasherPool = sync.Pool{
+	New: func() interface{} {
+		h, _ := blake2b.New256(nil)
+
+		return &Hasher{
+			tmp:  make(sliceBuffer, 0, 520), // cap is as large as a full branch node.
```

Comment:
how was this calculated? branch nodes can still store a value, so the cap might be larger than this

Reply:
I assumed it will be only keyed. I will look into this.

Reply:
let me know if you fix this and I can take another look